### DISCUSSION TITLE
Replicate Computerphile example

### DISCUSSION THREAD

**Discussion by isidroas** [Nikeshbajaj/Linear_Feedback_Shift_Register#9] [Q&A] [answered]
What arguments I need to pass to replicate the example shown in this video https://youtu.be/Ks1pw1X22y4?si=4bk5MuF5VXmgIhqL&t=109 ?

capture:
![2024-01-28-150357_](https://github.com/Nikeshbajaj/Linear_Feedback_Shift_Register/assets/10489506/ef7b4fab-c659-4a18-ab5b-3a1fc8f1f31e)

**Comment by Nikeshbajaj** [comment, marked as answer]
As video says, it is a 4-bit LFSR with tabs (or feedback polynomial ) as [4,3] (feeback from 4th and 3rd bit) with initial state as 1001, here is a code that generate exactly the same output as in video:

```
import pylfsr
from pylfsr import LFSR

state = [1,0,0,1]
fpoly = [4,3]
L = LFSR(initstate=state,fpoly=fpoly,counter_start_zero=False)
print('count \t state \t\t\toutbit \t seq')
print('-'*50)
for _ in range(16):
    print(L.count,L.state,'',L.outbit,L.seq,sep='\t')
    L.next()
print('-'*50)
print('Output: ',L.seq)
```

<img width="723" alt="image" src="https://github.com/Nikeshbajaj/Linear_Feedback_Shift_Register/assets/9660799/f251b4cb-43bc-4373-9bfd-fd10bb1966d0">


You could also visualise the same LFSR as 

```
L.Viz()
```
![image](https://github.com/Nikeshbajaj/Linear_Feedback_Shift_Register/assets/9660799/6fd80249-1357-4f45-a772-b1e414648240)

**Note** that feeback-bit **1** in above figure is **Not** the feedback computed from current state shown in figure (1100) but from the previuse state (1001).

**Comment by isidroas** [comment]
Thanks for the extended response!

-- 
Isidro Arias
